janitor/functions/sort_column_value_order.py

```diff
@@ -50,6 +51,7 @@ def sort_column_value_order(
         Dataframe, or if column_value_order dictionary is empty.
     :return: This function returns a Pandas DataFrame
     """
+    df = df.copy()
```

Comment:
A couple of improvements I think we can incorporate here:
- Since we are already doing a copy here, we can avoid the use of `new_df` later on, and just use `df` all the way.
- Actually I think this is one example where guard clauses really shine; -> I suggest we do all the relevant type & value checks (using our `check` and `check_column` util functions) and raise all the errors we can first, *before* doing the `df.copy()` and the rest of the sorting logic in an unnested block.
- And the `del new_df['cond_order']` line can be refactored out of the if-else block, to reduce code duplication.